h2o-core/src/main/java/water/fvec/RollupStats.java

```diff
@@ -217,6 +208,24 @@ else if (c instanceof C8DChunk)
     return this;
   }
 
+  private long processUUIDchunk(Chunk c, long checksum, long start, long l) {
+    for( int i=c.nextNZ(-1); i< c._len; i=c.nextNZ(i) ) {
+      if( c.isNA(i) ) _naCnt++;
+      else {
+        try {
+          long lo = c.at16l(i), hi = c.at16h(i);
+          if (lo != 0 || hi != 0) _nzCnt++;
+          l = lo ^ 37*hi;
+        } catch (Chunk.WrongType wt) {
```

Comment:
Your change in NewChunk doesn't let the chunk to be created if detected and declared types differ (in UUID case). This is a good change. If this constraint is violated an exception is thrown.

I guess my question is: what will happen if we try to reproduce the reported issue? This pull request makes me believe that it will result in an exception - there is no code that prevents that. Is that a correct understanding?